stories/widgets.stories.js

```diff
@@ -131,3 +169,31 @@ storiesOf( 'Global/Widgets', module )
 			) ) }
 		</CompositeWidgets>
 	) );
+
+storiesOf( 'Global/Widgets/Widget Area', module )
+	.addDecorator( ( storyFn ) => storyFn( createTestRegistry() ) )
+	.add( 'Regular sizes', ( registry ) => (
+		<WithTestRegistry registry={ registry }>
+			{ createWidgetAreas(
+				registry,
+				[ QUARTER, QUARTER, QUARTER, QUARTER ],
+				[ HALF, QUARTER, QUARTER ],
+				[ QUARTER, HALF, QUARTER ],
+				[ QUARTER, QUARTER, HALF ],
+				[ HALF, HALF ],
+				[ FULL ],
+			) }
+		</WithTestRegistry>
+	) )
+	.add( 'Irregular sizes', ( registry ) => (
+		<WithTestRegistry registry={ registry }>
+			{ createWidgetAreas(
+				registry,
+				[ QUARTER, QUARTER, HALF, [ QUARTER, FULL ] ],
+				[ [ HALF, FULL ], QUARTER, QUARTER ],
+				[ QUARTER, [ FULL, HALF ], QUARTER ],
+				[ QUARTER, QUARTER, [ HALF, FULL ] ],
```

Comment:
The overall implementation of these stories looks good, however a few "key sizes" to test some behavior is missing (those are outlined in the IB):
* 3 quarter widgets (tests that expanding them equally to fill the full width works, see the `counter === 9` logic)
* 1 half widget, 1 quarter widget that has half as secondary width, 1 full widget (tests that filling up remaining space in a row with the fallback width works, see my above comment where the previous widget should be checked to find the maximum width that is <= 12)

Reply:
Added these variations to the storybook story.